subsys/bluetooth/audio/ascs.c

```diff
@@ -3080,22 +3088,109 @@ static ssize_t ascs_cp_write(struct bt_conn *conn,
+	LISTIFY(CONFIG_BT_ASCS_MAX_ASE_SNK_COUNT, BT_ASCS_ASE_SNK_DEFINE, (,)),
+#else
+#define BT_ASCS_ASE_SINKS
+#endif /* CONFIG_BT_ASCS_MAX_ASE_SNK_COUNT > 0 */
+
+#if CONFIG_BT_ASCS_MAX_ASE_SRC_COUNT > 0
+#define BT_ASCS_ASE_SOURCES \
+	LISTIFY(CONFIG_BT_ASCS_MAX_ASE_SRC_COUNT, BT_ASCS_ASE_SRC_DEFINE, (,)),
+#else
+#define BT_ASCS_ASE_SOURCES
+#endif /* CONFIG_BT_ASCS_MAX_ASE_SRC_COUNT > 0 */
+
+#define BT_ASCS_SERVICE_DEFINITION() { \
+	BT_GATT_PRIMARY_SERVICE(BT_UUID_ASCS), \
+	BT_AUDIO_CHRC(BT_UUID_ASCS_ASE_CP, \
+		      BT_GATT_CHRC_WRITE | BT_GATT_CHRC_WRITE_WITHOUT_RESP | BT_GATT_CHRC_NOTIFY, \
+		      BT_GATT_PERM_WRITE_ENCRYPT | BT_GATT_PERM_PREPARE_WRITE, \
+		      NULL, ascs_cp_write, NULL), \
+	BT_AUDIO_CCC(ascs_cp_cfg_changed), \
+	BT_ASCS_ASE_SINKS \
+	BT_ASCS_ASE_SOURCES \
+	}
+
+#define ASCS_ASE_CHAR_ATTR_COUNT 3 /* declaration + value + cccd */
+
+static struct bt_gatt_attr ascs_attrs[] = BT_ASCS_SERVICE_DEFINITION();
+static struct bt_gatt_service ascs_svc = (struct bt_gatt_service)BT_GATT_SERVICE(ascs_attrs);
+
+static void configure_ase_char(uint8_t snk_cnt, uint8_t src_cnt)
+{
+	uint8_t snk_ases_to_rem = CONFIG_BT_ASCS_MAX_ASE_SNK_COUNT - snk_cnt;
+	uint8_t src_ases_to_rem = CONFIG_BT_ASCS_MAX_ASE_SRC_COUNT - src_cnt;
+	size_t attrs_to_rem;
+
+	/* Remove the Source ASEs. The ones to remove will always be at the very tail of the
+	 * attributes, so we just decrease the count withe the amount of sources we want to remove.
+	 */
+	attrs_to_rem = src_ases_to_rem * ASCS_ASE_CHAR_ATTR_COUNT;
+	ascs_svc.attr_count -= attrs_to_rem;
+
+	/* Remove the Sink ASEs.
+	 */
+	attrs_to_rem = snk_ases_to_rem * ASCS_ASE_CHAR_ATTR_COUNT;
+	if (CONFIG_BT_ASCS_MAX_ASE_SRC_COUNT == 0 || CONFIG_BT_ASCS_MAX_ASE_SRC_COUNT
+	    == src_ases_to_rem) {
+		/* If there are no Source ASEs present, then we can just decrease the
+		 * attribute count
+		 */
+		ascs_svc.attr_count -= attrs_to_rem;
+	} else {
+		/* As Source ASEs are present, we need to iterate backwards (as this will likely be
+		 * the shortest distance). Find the first Sink to save, and move all Sources
+		 * backwards to it.
+		 */
+		size_t src_start_idx = ascs_svc.attr_count - (src_cnt * ASCS_ASE_CHAR_ATTR_COUNT);
+		size_t new_src_start_idx = src_start_idx - (snk_ases_to_rem *
+							    ASCS_ASE_CHAR_ATTR_COUNT);
+
+		for (size_t i = 0; i < src_cnt * ASCS_ASE_CHAR_ATTR_COUNT; i++) {
+			ascs_svc.attrs[new_src_start_idx + i] = ascs_svc.attrs[src_start_idx + i];
+		}
+
+		ascs_svc.attr_count -= attrs_to_rem;
+	}
+}
+
+int bt_ascs_register(uint8_t snk_cnt, uint8_t src_cnt)
+{
+	int err = 0;
+
+	if (ascs.registered) {
+		LOG_ERR("ASCS already registered");
+
+		return -EALREADY;
+	}
+
+	if (snk_cnt > CONFIG_BT_ASCS_MAX_ASE_SNK_COUNT ||
+	    src_cnt > CONFIG_BT_ASCS_MAX_ASE_SRC_COUNT) {
+		//LOG_ERR;
+		return -EINVAL;
+	}
+	configure_ase_char(snk_cnt, src_cnt);
+
+	err = bt_gatt_service_register(&ascs_svc);
+	if (err != 0) {
+		LOG_ERR("Failed to register ASCS in gatt DB");
+	}
```

Comment:
```suggestion
	err = bt_gatt_service_register(&ascs_svc);
	if (err != 0) {
		LOG_DBG("Failed to register ASCS in gatt DB");
	}
```

Suggest not to use `LOG_ERR` for functions that return the error code to the application

Reply:
Good point. Also, this should return the error code here.